### PR TITLE
Fix IndexError when using bitbucket

### LINKED ISSUE #723: Crash in git module
I was pushing a branch, and executed `fuck` afterwards. Here is the stacktrace:

```Traceback (most recent call last):
  File "/usr/local/bin/thefuck", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/dist-packages/thefuck/entrypoints/main.py", line 26, in main
    fix_command(known_args)
  File "/usr/local/lib/python3.5/dist-packages/thefuck/entrypoints/fix_command.py", line 42, in fix_command
    selected_command = select_command(corrected_commands)
  File "/usr/local/lib/python3.5/dist-packages/thefuck/ui.py", line 71, in select_command
    selector = CommandSelector(corrected_commands)
  File "/usr/local/lib/python3.5/dist-packages/thefuck/ui.py", line 34, in __init__
    self._commands = [next(self._commands_gen)]
  File "/usr/local/lib/python3.5/dist-packages/thefuck/corrector.py", line 43, in organize_commands
    first_command = next(corrected_commands)
  File "/usr/local/lib/python3.5/dist-packages/thefuck/corrector.py", line 74, in <genexpr>
    for corrected in rule.get_corrected_commands(command))
  File "/usr/local/lib/python3.5/dist-packages/thefuck/types.py", line 189, in get_corrected_commands
    new_commands = self.get_new_command(command)
  File "<decorator-gen-104>", line 2, in get_new_command
  File "/usr/local/lib/python3.5/dist-packages/thefuck/specific/git.py", line 32, in git_support
    return fn(command)
  File "/usr/local/lib/python3.5/dist-packages/thefuck/rules/git_push.py", line 36, in get_new_command
    arguments = re.findall(r'git push (.*)', command.output)[0].strip()
IndexError: list index out of range
```

This is the output from the command before (which was only `git push`):
```git push --set-upstream origin feature/5_exec_command [enter/↑/↓/ctrl+c]
Total 0 (delta 0), reused 0 (delta 0)
remote: 
remote: Create pull request for feature/[...]:
remote:   https://bitbucket.org/[...]
remote: 
To git@bitbucket.org:[...].git
   e5e7fbb..700d998  feature/[...] -> feature/[...]
Branch feature/[...] set up to track remote branch feature/[...] from origin.
```

### PR DESCRIPTION
**Pull request by davidhart82**
This should fix #723.

### CODE REVIEW

File: thefuck/rules/git_push.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 @git_support
 def match(command):
     return ('push' in command.script
-            and 'set-upstream' in command.output)
+            and 'git push --set-upstream' in command.output)
```
Comment (josephfrazier) [inline thefuck/rules/git_push.py]: I'm not sure how this change relates to the tests that were added. If I undo the changes on this line, the new tests still pass. Can you help me understand how this fixes the issue?

Comment (davidhart82) [reply thefuck/rules/git_push.py]: I added a test to see if we would match on the bitbucket output, which suggests we won't - but thought it best to leave it.  I can only assume that the output provided was not the entire output, but the only text checked for was `set-upstream`.

Change #682 modified git_push.py to regex search for `git push`, assuming it was there.  You can't do that when the only thing we ever check for in the output is `set-upstream`.

Comment (davidhart82) [reply thefuck/rules/git_push.py]: I guess a much better example might be if the branch was called `feature/set-upstream`.

File: thefuck/rules/git_push.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 @git_support
 def match(command):
     return ('push' in command.script_parts
-            and 'set-upstream' in command.output)
+            and 'git push --set-upstream' in command.output)
```
Comment (josephfrazier) [inline thefuck/rules/git_push.py]: I'm thinking we should just make this line be:

```python
            and '--set-upstream' in command.script_parts)
```

in order to allow `--set-upstream` to occur elsewhere than right after `push`. Unless someone names a branch `--set-upstream`, this should be fine.
